### PR TITLE
Adding choice plugin !choix

### PR DESCRIPTION
**Pull request by jiquiame**
Adding the super new command !choix allowing to make enlightened decisions directly using the bot. Supports two words + a bunch of separators

### CODE REVIEW

File: src/coucoubot.ml [hunk 1]
```diff
@@ -22,10 +23,10 @@ let config = {
   C.Config.
   server = "irc.freenode.net";
   port = 7000;
-  username = "coucoubot";
-  realname = "coucoubot";
-  nick = "coucoubot";
-  channel = "#arch-fr-free";
+  username = "roucoubot";
```
Comment (c-cube) [inline src/coucoubot.ml]: heh, don't push your test setup here :laughing:

File: src/Plugin_choice.ml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+open Prelude
+
+(** Separate a string into two parts using the given separator **)
+let extract_choices sep str =
+  let rec list_to_string l =
+    match l with
+      | s::ss -> s ^ " " ^ list_to_string ss
+      | [] -> ""
+  in
+  let rec remove_empty_strings l =
```
Comment (c-cube) [inline src/Plugin_choice.ml]: `List.filter (fun s -> s <> "")` would do

**Review by c-cube — COMMENTED** [review]
Looks good, just revert the test config and it'll be a :heavy_check_mark:

File: src/coucoubot.ml (hunk 1 above)
Comment (jiquiame) [reply src/coucoubot.ml]: Uh, thought I correctly add -p'd it :(

File: src/Plugin_choice.ml (hunk 2 above)
Comment (jiquiame) [reply src/Plugin_choice.ml]: Much better indeed

**Pull request merged**